verificationtool/src/main/java/com/greenops/verificationtool/datamodel/verification/DAG.java

```diff
@@ -0,0 +1,287 @@
+package com.greenops.verificationtool.datamodel.verification;
+
+import com.greenops.util.datamodel.auditlog.DeploymentLog;
+import com.greenops.util.datamodel.auditlog.Log;
+import com.greenops.util.datamodel.pipelinestatus.PipelineStatus;
+import com.greenops.verificationtool.datamodel.pipelinedata.PipelineData;
+import com.greenops.verificationtool.datamodel.pipelinedata.StepData;
+
+import java.util.*;
+import com.greenops.util.datamodel.event.*;
+import com.greenops.verificationtool.datamodel.pipelinedata.Test;
+
+public class DAG {
+    private final String PipelineTriggerEvent = "PipelineTriggerEvent";
+    private final String TriggerStepEvent = "TriggerStepEvent";
+    private final String ApplicationInfraCompletionEvent = "ApplicationInfraCompletionEvent";
+    private final String ApplicationInfraTriggerEvent = "ApplicationInfraTriggerEvent";
+    private final String ClientCompletionEvent = "ClientCompletionEvent";
+    private final String TestCompletionEvent = "TestCompletionEvent";
+    private PipelineData pipelineData;
+    private String pipelineName;
+    private HashMap<Vertex, List<Vertex>> graph;
+    private HashMap<Vertex, List<Vertex>> reverseGraph;
+    private Vertex root;
+    private HashMap<Vertex, Boolean> visited;
+    private List<Vertex> ancestorsList;
+
+    public DAG(PipelineData pipelineData, String pipelineName){
+        this.pipelineData = pipelineData;
+        this.pipelineName = pipelineName;
+        this.graph = new HashMap<Vertex, List<Vertex>>();
+        this.reverseGraph = new HashMap<Vertex, List<Vertex>>();
+        this.visited = new HashMap<Vertex, Boolean>();
+
+        this.createDAG();
+    }
+
+    private void createDAG(){
+        Vertex pipelineTriggerEventVertex = new Vertex(this.PipelineTriggerEvent, this.pipelineName, "ATLAS_ROOT_DATA");
+        this.root = pipelineTriggerEventVertex;
+        this.graph.put(pipelineTriggerEventVertex, new ArrayList<Vertex>());
+
+        List<String> allStepNames = this.pipelineData.getAllSteps();
+        List<String> rootSteps = this.getRootSteps(allStepNames);
+
+        StepOrderGraph stepOrderGraph = new StepOrderGraph(this.pipelineData);
```

Comment:
this graph is being used for building whole DAG. I'm first creating a dummy graph of only step names, so that I can add edges between two different steps. Basically, this will let us know, which step will be after which step.